lib/ui/painting.dart

```diff
@@ -3612,6 +3612,183 @@ class PictureRecorder extends NativeFieldWrapperClass2 {
+  /// To honor this as well, the shape should be translated by [offset] before
+  /// being filled using this [Paint].
+  Paint toPaint({bool debugDisableShadows = false}) {
+    final Paint result = Paint()
+      ..color = color
+      ..maskFilter = MaskFilter.blur(BlurStyle.normal, blurSigma);
+    return result;
+  }
+
+  /// Returns a new box shadow with its offset and blurRadius scaled by the given factor.
+  Shadow scale(double factor) {
+    return Shadow(
+      color: color,
+      offset: offset * factor,
+      blurRadius: blurRadius * factor,
+    );
+  }
+
+  /// Linearly interpolate between two shadows.
+  ///
+  /// If either shadow is null, this function linearly interpolates from a
+  /// a shadow that matches the other shadow in color but has a zero
+  /// offset and a zero blurRadius.
+  ///
+  /// {@macro flutter.material.themeData.lerp}
+  static Shadow lerp(Shadow a, Shadow b, double t) {
+    assert(t != null);
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return b.scale(t);
+    if (b == null)
+      return a.scale(1.0 - t);
+    return Shadow(
+      color: Color.lerp(a.color, b.color, t),
+      offset: Offset.lerp(a.offset, b.offset, t),
+      blurRadius: lerpDouble(a.blurRadius, b.blurRadius, t),
+    );
+  }
+
+  /// Linearly interpolate between two lists of shadows.
+  ///
+  /// If the lists differ in length, excess items are lerped with null.
+  ///
+  /// {@macro flutter.material.themeData.lerp}
+  static List<Shadow> lerpList(List<Shadow> a, List<Shadow> b, double t) {
+    assert(t != null);
+    if (a == null && b == null)
+      return null;
+    a ??= <Shadow>[];
+    b ??= <Shadow>[];
+    final List<Shadow> result = <Shadow>[];
+    final int commonLength = math.min(a.length, b.length);
+    for (int i = 0; i < commonLength; i += 1)
+      result.add(Shadow.lerp(a[i], b[i], t));
+    for (int i = commonLength; i < a.length; i += 1)
+      result.add(a[i].scale(1.0 - t));
+    for (int i = commonLength; i < b.length; i += 1)
+      result.add(b[i].scale(t));
+    return result;
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other is! Shadow)
+      return false;
+    final Shadow typedOther = other;
+    return color == typedOther.color &&
+      offset == typedOther.offset &&
+      blurRadius == typedOther.blurRadius;
+  }
+
+  /// Determines if lists [a] and [b] are equivalent. Handles all cases of null/non-null
+  /// values of a and b.
+  static bool shadowsListEquals(List<Shadow> a, List<Shadow> b) {
+    // Compare _shadows
+    if (a == null && b == null)
+      return true;
+    if (a != null && b == null ||
+      a == null && b != null)
+      return false;
+    if (a.length != b.length)
+      return false;
+    for (int index = 0; index < a.length; ++index)
+      if (a[index] != b[index])
+        return false;
+    return true;
+  }
```

Comment:
this should be private. we want most people to use the generic listEquals from the foundation library.